index.js

```diff
@@ -58,6 +58,78 @@ const main = async (options = {}) => {
 		}));
 };
 
-module.exports = process.platform === 'win32' ? windows : main;
+const ERROR_MSG_PARSING_FAILED = 'ps output parsing failed';
+
+const psFields = 'pid,ppid,uid,%cpu,%mem,comm,args';
+// TODO Use named capture groups when targeting Node.js 10+
+const psOutputRegex = /^[ \t]*(\d+)[ \t]+(\d+)[ \t]+(\d+)[ \t]+(\d+\.\d+)[ \t]+(\d+\.\d+)[ \t]+/; // Groups: pid, ppid, uid, cpu, mem
+
+const nonWindowsSingleCall = async (options = {}) => {
+	const flags = options.all === false ? 'wwxo' : 'awwxo';
+
+	const [psPid, stdout] = await new Promise((resolve, reject) => {
+		const child = childProcess.execFile('ps', [flags, psFields], {maxBuffer: TEN_MEGABYTES}, (error, stdout) => {
```

Comment:
This should use the promise version of execFile, see https://github.com/sindresorhus/ps-list/blob/c6d4fd17f6c088b09760765939d71d8674f13301/index.js#L7

Reply:
We need to get the pid here. so have to have access not only to promise result but to sync function result too.

Reply:
Oh, I kinda just assumed the promise version of `execFile` would include that, but turns out not: https://github.com/nodejs/node/issues/28244 ...